### ISSUE TITLE
"Remove unnecessary braces" removes `self as name` imports

### ISSUE THREAD

**Issue by lowr** [rust-lang/rust-analyzer#15191]
```rust
use my_awesome_mod::{self as cool_name};
  //^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Quick fix: Remove unnecessary braces
```

after applying the quick fix:

```rust
use my_awesome_mod;
```

The quick fix should be:

```rust
use my_awesome_mod as cool_name;
```

**rust-analyzer version**: 3d3f325a4ec2

**rustc version**: 1.70

**relevant settings**: (eg. client settings, or environment variables like `CARGO`, `RUSTC`, `RUSTUP_HOME` or `CARGO_HOME`)

**Comment by lowr** [comment]
Maybe we shouldn't show the diagnostic for `self` imports at all because `use a::{self};` and `use a;` are semantically different.
